graphql-language/src/ast.rs

```diff
@@ -7,22 +7,23 @@ use source::Source;
 
 /// Contains some character offsets that identify where the source of the AST
 /// is from.
+#[derive(Clone, Copy)]
 pub struct Location<'a> {
   pub start: usize,
   pub end: usize,
-  pub source: Option<&'a Source<'a>>
+  pub source: Source<'a>
 }
 
 /// All AST node types implement this trait.
 pub trait Node<'a> {
-  fn location(&self) -> Option<&Location<'a>>;
```

Comment:
In the reference implementation, `Location` is always optional. I can see an argument for making it required, however the AST may not always be parsed from a source. What about when the AST is manually constructed? In testing for example. Will we have to create fake `Location`s?